ngraph_creator/operations/src/ROI_Pooling.cpp

```diff
@@ -0,0 +1,107 @@
+//#define LOG_NDEBUG 0
+#include <ROI_Pooling.hpp>
+#define LOG_TAG "ROI_Pooling"
+
+namespace android {
+namespace hardware {
+namespace neuralnetworks {
+namespace nnhal {
+
+ROI_Pooling::ROI_Pooling(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+bool ROI_Pooling::validate() {
+    ALOGV("%s Entering", __func__);
+
+    // Check Output type
+    if (!checkOutputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32)) {
+        return false;
+    }
+
+    // Check Input Type
+    if (!checkInputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32)) return false;
+    if (!checkInputOperandType(1, (int32_t)OperandType::TENSOR_FLOAT32)) return false;
+
+    if (!checkInputOperandType(2, (int32_t)OperandType::TENSOR_INT32)) return false;
+    if (!checkInputOperandType(3, (int32_t)OperandType::INT32)) return false;
+    if (!checkInputOperandType(4, (int32_t)OperandType::INT32)) return false;
+    if (!checkInputOperandType(5, (int32_t)OperandType::FLOAT32)) return false;
+    if (!checkInputOperandType(6, (int32_t)OperandType::FLOAT32)) return false;
+    if (!checkInputOperandType(7, (int32_t)OperandType::BOOL)) return false;

```

Comment:
Do we really need to verify the data type of all scalars?

Reply:
Are we not suppose to? because I see most of the ops did check.

Reply:
In both getsupportedops and preparedmodel calls, we are calling validatemodel(). ValidateModel function checks for input and output data type checks as part of validateOperations. If the checks are already performed there, does it make sense to add the check here too?

Can u verify whether validate() function for ROIPooling gets called during validateModel call?? If so ROIPooling validate already checks the input and output data types.

Reply:
Just to be clear.. validate() function is implemented under frameworks/ml/nn/common/operations/RoiPooling.cpp. Which will be called when validateModel gets called.. If this is not getting called, then we should keep the above check..

Reply:
CTS/VTS passes even without checks

Reply:
I see the validate() function is getting called when validateModel is getting called